### DISCUSSION TITLE
We can have Prometheus metrics

### DISCUSSION THREAD

**Discussion by EngRajabi** [ThreeMammals/Ocelot#1827] [Ideas]
In my opinion, this beautiful gateway lacks a metric to monitor. It is very easy to produce Prometheus metrics that are supported by ready-made dashboards.
Many times it happens that the services are interrupted or there is a problem with the gateway connection, with these metrics you can quickly find out.
In addition to these metrics, it gives us very useful information about the amount of resources consumed by the gateway.
We can use the link below
https://github.com/prometheus-net/prometheus-net

**Comment by ggnaegi** [comment]
@raman-m @RaynaldM I agree, it would be a nice addition.

**Reply by RaynaldM** [reply]
Yes, sure, we should implement OpenTelemetry middleware.

**Reply by EngRajabi** [reply]
@RaynaldM 
I suggest not to use the OpenTelemetry package directly for several reasons
- There is still an RC version
- Performance is lower than Prometheus package

**Reply by ggnaegi** [reply]
What if we collect metrics using EventSource and then we provide interfaces for the consumers?

**Reply by raman-m** [reply]
Ideas ideas ideas... 😄 
Where is our lonely senior contributor for these ideas? 🤣

**Reply by raman-m** [reply]
@EngRajabi Mohsen, Raynald's team will develop this OpenTelemetry module for monitoring.

Right, @RaynaldM ? 😄

**Reply by RaynaldM** [reply]
right

**Comment by raman-m** [comment]
>@EngRajabi [on Nov 30](#discussion-5909189)

Hello Mohsen!
Good idea. Prometheus is the perfect monitoring tool which was/is quite popular. And Ocelot has no Monitoring feature at all in comparison to other popular gateways...
Will you or your team to contribute to that Monitoring feature?

Also, I'm just curious... In what industry your (your team) software uses Ocelot as a product?

**Reply by EngRajabi** [reply]
Hello, our product is for the stock market and banks.
https://www.emofid.com/ (The largest stock brokerage in Iran)
We use Ocelot.
No need to implement OpenTelemetry. I can add the work we did in our company for Monitoring to Ocelot. I will do this with the help of the package https://github.com/prometheus-net/prometheus-net and less code.

**Comment by EngRajabi** [comment]
Should I add Prometheus with a PR?

**Reply by raman-m** [reply]
I have no time for such development adventures... Sorry!

**Reply by raman-m** [reply]
Please, help us with current releases, and we will brainstorm this Idea once again continuing the discussion.
I know you have developers in your current team.

**Reply by EngRajabi** [reply]
If you agree, I will personally add this feature to the project.

**Reply by raman-m** [reply]
@EngRajabi Mohsen, to be fair...
Your developers from your development team did a lot of design issues being incidentally merged by previous repo maintainers. As a team, we have to review & re-design them, review and redesign...
I don't what to mentor you... Maybe @ggnaegi will mentor you?! But I'm not sure...
As a compromise our team could let you create PR, then I will mark it as draft to start discussions...
But it is better to discuss here before PR creation.

@ggnaegi  FYI
